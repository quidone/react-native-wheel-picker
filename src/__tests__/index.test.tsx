import WheelPicker from '@implementation/base';
import React from 'react';
import renderer from 'react-test-renderer';

describe('WheelPicker', () => {
  it('should match snapshot', () => {
    const tree = renderer
      .create(
        <WheelPicker
          data={[{value: 1, label: 'Item 1'}]}
          testID="wheel-picker"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
