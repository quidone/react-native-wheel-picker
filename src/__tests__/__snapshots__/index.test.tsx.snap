// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WheelPicker should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
      Object {
        "height": 240,
        "width": "auto",
      },
    ]
  }
  testID="wheel-picker"
>
  <RCTScrollView
    collapsable={false}
    contentContainerStyle={
      Array [
        Object {
          "paddingVertical": 96,
        },
        undefined,
      ]
    }
    contentOffset={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    nestedScrollEnabled={true}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollEndDrag={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    removeClippedSubviews={false}
    scrollEnabled={true}
    scrollEventThrottle={16}
    showsHorizontalScrollIndicator={false}
    showsVerticalScrollIndicator={false}
    snapToOffsets={
      Array [
        0,
      ]
    }
    style={
      Object {
        "overflow": "visible",
        "width": "100%",
      }
    }
    visibleItemCount={5}
  >
    <View>
      <View
        collapsable={false}
        style={
          Object {
            "height": 48,
            "opacity": 1,
            "transform": Array [
              Object {
                "translateY": 0,
              },
              Object {
                "rotateX": "0deg",
              },
              Object {
                "perspective": 1000,
              },
            ],
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 20,
                "textAlign": "center",
              },
              Object {
                "lineHeight": 48,
              },
              undefined,
            ]
          }
        >
          Item 1
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "alignItems": "center",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#000",
            "borderRadius": 8,
            "opacity": 0.05,
          },
          Object {
            "height": 48,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
